proposals/shared-everything-threads/Overview.md

```diff
@@ -116,12 +116,33 @@ meaning `shared` or `unshared` introduced in the shared memory proposal.
 
 Shared memories have already been standardized.
 
+For consistency with other shared annotations, the existing text format for shared memories is
+redefined to be an abbreviation of the new text format:
+
+```
+memtype ::= it:index_type lim:limits => unshared it lim
+          | '(' 'shared' it:index_type lim:limits ') => shared it lim
+
+'(' 'memory' id? index_type limits 'shared' ')'
+  == '(' 'memory' id?  '(' 'shared' index_type limits ')' ')'
+
+'(' 'memory' id? '(' 'import' name1 name2 ')' index_type limits 'shared' ')'
+  == '(' 'memory' id? '(' 'import' name1 name2 ')' '(' 'shared' index_type limits ')' ')'
+```
+
 #### Tables
 
-The syntax of `tabletype` is extended:
+The abstract syntax of `tabletype` is extended:
+
+```
+tabletype ::= share index_type limits reftype
+```
+
+Similarly, the text format is extended:
 
 ```
-tabletype ::= share limits reftype
+tabletype ::= it:index_type lim:limits rt:reftype => unshared it lim rt
+            | '(' 'shared' it:index_type lim:limits rt:reftype ')' => shared it lim rt
```

Comment:
Seeing this, it's odd that we demand outer parens here when shared and none otherwise (probably historical accident). Let's just drop them here and same for memory types.

Reply:
I've been looking at this for a while and I've gotten used to seeing `(shared any)` and `any`, e.g., for heap types; it makes sense to have "one thing" written without parentheses but "multiple things" grouped with parentheses. It seems like we should be consistent with the heap type text format and keep the parentheses here.

Reply:
Yeah, that also seems like a more consistent option...

Reply:
My comment only concerns memory/table types, and the introduction of _outer_ parens there, which for some reason they never had. Point being that it would be a bit odd to have to write, e.g.,
```
(table 0 10 funcref)
```
and at the same time
```
(table (shared 0 10 funcref))
```
The situation would be different had the former been
```
(table (0 10 funcref))
```
because then we could naturally make the latter
```
(table (shared (0 10 funcref)))
```
But for better or worse, that's not our starting point. (Historically I believe that is because we originally did not think of the extra fields on table/memory declarations as types in their own right.)

FWIW, this is unrelated to the question whether shared comes first or last.

Reply:
I see what you mean when I think about consistency with globals. Since we currently write `(global (mut i32) ...)` I went with `(global (shared mut i32) ...)`. But you're pointing out (i.e., "historical accident") that the text format is already inconsistent between globals and memories/tables. If globals already get parentheses and memories/tables don't, then I agree we shouldn't add parentheses to memories and tables even if they become shared.

(But is there no way to fix this historical inconsistency? Maybe it's too late or this proposal is not the right place?).

Reply:
Yeah, I don't really see a difference between having `i32` and `(mut i32)` for globals and having `0 10 funcref` and `(shared 0 10 funcref)` for tables.

But I'll upload a commit that gets rid of the parens and we can see what we think about that.